### PR TITLE
Add TranslateOffsets RPC

### PR DESCRIPTION
**Pull request by scanterog**
This PR adds a new TranslateOffsets RPC. This endpoints relies on mirrorMaker2 that periodically emit checkpoints in the destination cluster (on a topic), containing offsets for each consumer group in the source cluster. The MirrorCheckpointConnector will periodically query the source cluster for all committed offsets from all consumer groups, filter for those topics being replicated, and emit a message to a topic like `source.checkpoints.internal` in the destination cluster.

This RPC consumes this particular topic in the destination cluster and based on the source cluster alias and the consumer group name, returns the upstream offset and its corresponding local offset.

Query example: `source` is the source cluster alias and `sam` is the consumer group id.
```
curl localhost:8080/v1/translate-offsets/source/sam
{
  "offsets": {
    "source.foo-1:0": {
      "upstream_offset": "3",
      "local_offset": "3"
    },
    "source.foo-2:0": {
      "upstream_offset": "5",
      "local_offset": "5"
    },
    "source.foo-2:1": {
      "upstream_offset": "3",
      "local_offset": "3"
    },
    "source.foo-2:2": {
      "upstream_offset": "7",
      "local_offset": "7"
    }
  }
}

```

### CODE REVIEW

File: registry/server/api_offset_translation.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+		return nil, err
+	}
+	err = decodeKey(msg.Key, &checkpoint)
+	if err != nil {
+		return nil, err
+	}
+	return &checkpoint, nil
+}
+
+func decodeValue(record []byte, checkpoint *Checkpoint) error {
```
Comment (Mwea) [inline registry/server/api_offset_translation.go:128]: You might want to test this

File: registry/server/api_offset_translation.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+	if strLength == 0 {
+		checkpoint.Metadata = ""
+	} else {
+		checkpoint.Metadata = string(record[20 : 20+strLength])
+	}
+
+	return nil
+}
+
+func decodeKey(record []byte, checkpoint *Checkpoint) error {
```
Comment (Mwea) [inline registry/server/api_offset_translation.go:146]: same you might want to test this

File: registry/server/api_offset_translation.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+	s.kafkaconsumer.Assign(checkpointAssignment)
+	s.kafkaconsumer.Seek(checkpointTopicPartition, seekTimeoutMs)
+	_, endOffset, err := s.kafkaconsumer.QueryWatermarkOffsets(
+		checkpointTopic, checkpointPartition, queryWatermarkOffsetsTimeoutMs,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve the log end offset for %s:%d", checkpointTopic, checkpointPartition)
+	}
+
+	endOfStream := func() bool {
```
Comment (Mwea) [inline registry/server/api_offset_translation.go:82]: Wow. that's a fancy way to do a loop

**Review by Mwea — APPROVED** [review]
LGTM overall 🔥  I would potentially add some tests to ensure it behaves as expected as this service will be used for very critical use-cases

**Review by jamiealquiza — APPROVED** [review]

**Pull request merged**